src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/CopiedFieldRvaNode.cs

```diff
@@ -76,18 +75,20 @@ private unsafe byte[] GetRvaData(int targetPointerSize)
                 int currentSize = field.FieldType.GetElementSize().AsInt;
                 if (currentSize > size)
                 {
-                    if (currentSize > memBlock.Length)
-                        throw new BadImageFormatException();
-
                     // We need to handle overlapping fields by reusing blobs based on the rva, and just update
                     // the size and contents
                     size = currentSize;
                     result = new byte[AlignmentHelper.AlignUp(size, targetPointerSize)];
```

Comment:
We can also move the array allocation outside of the loop. No need to allocate and resize now